repl: use better uncaught exceptions indicator

BridgeAR opened the pull request:
This switches "Thrown:" with "Uncaught" to outline clearer that the
thrown error is not caught. It will also mark the error in red in
case the terminal supports colors / if colors are activated for the
repl.

Since this should only be relevant for the end user, should this be semver-major or is it fine as semver-minor?

**Update**: I removed the colors as requested below!

Before
![image](https://user-images.githubusercontent.com/8822573/65701402-8186aa00-e081-11e9-97dd-f42d8416fa8f.png)
![image](https://user-images.githubusercontent.com/8822573/65701410-88adb800-e081-11e9-964d-b8c913d9b427.png)

After (with different color themes and on different terminals)
![image](https://user-images.githubusercontent.com/8822573/68930774-66860b80-078f-11ea-878f-0db5cfcfacaf.png)
![image](https://user-images.githubusercontent.com/8822573/68937356-82dc7500-079c-11ea-9bc6-59671902f0b1.png)
![image](https://user-images.githubusercontent.com/8822573/68937376-8cfe7380-079c-11ea-98ca-5bdfd9c2e2ab.png)
![image](https://user-images.githubusercontent.com/8822573/68937397-9982cc00-079c-11ea-87d2-f7c71895a367.png)
![image](https://user-images.githubusercontent.com/8822573/68937454-b61f0400-079c-11ea-9886-40436527e334.png)

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

BridgeAR commented:
@nodejs/repl PTAL

jasnell commented:
Generally fine with this approach but the change in output *may* make this semver-major :-/

BridgeAR commented:
@nodejs/tsc PTAL and provide your opinion about the semverness. It is not immediately clear what this should ideally be.

BridgeAR commented:
I added some before and after screenshots in the description above.

jasnell commented:
btw, fwiw I *really* dislike having to potentially treat these kinds of usability improvement changes to console output as semver-major but we have seen in the past that such changes can break existing code. I would be definitely amenable to any policy change or approach to handling these things that would allow us to avoid treating them as semver-major

Trott commented:
Maybe a CITGM run will help make a decision for us? I wouldn't expect this in particular to break anything in CITGM, but if it does, that's certainly an argument for semver-major.

devsnek commented:
i like changing it to uncaught, not a huge fan of the red though. red kills readability on pretty much everything that isn't green thanks to the magic of our eyes and rgb.

Trott commented:
Needs a rebase. Might make sense to split the wording change and the color change into separate PRs so any controversy about one won't delay the other.

lholmquist commented:
Agree that these are 2 separate concerns, and would be nice if they were broken into different PRs

BridgeAR commented:
I would prefer to keep the change together and not splitting this up.

I updated the colors to use bold bright red. This seems significantly better readability wise.

Before:
![image](https://user-images.githubusercontent.com/8822573/65701344-6fa50700-e081-11e9-99b5-73ab7922ce78.png)

After:
![image](https://user-images.githubusercontent.com/8822573/68930774-66860b80-078f-11ea-878f-0db5cfcfacaf.png)

I also added further screenshots at the top where I compare different color themes and terminals.

@devsnek @lholmquist PTAL

BridgeAR commented:
@nodejs/repl @nodejs/util PTAL

antsmartian commented:
I feel red color (with bold) is actually making it harder to read (on black terminals), IHO.

devsnek commented:
Can the colors be a separate PR? I'd love for the uncaught stuff to get in since that seems pretty uncontroversial.

BridgeAR commented:
@devsnek @antsmartian @lholmquist does either of you have a suggestion what colors might be good to use instead? I will separate the colors from the rest of the PR even though I personally think they belong somewhat together.

Right now it's not immediately clear when an error is thrown and when not. Using a color indicator overcomes that problem. Thus, I expect most users to profit from the current color suggestion. We could also use a predefined background color, I personally do not like that idea all that much though.

BridgeAR commented:
I just pushed another commit that removes the colors.

Trott commented:
If we have informed volunteers to do so, probably a good idea to get review by some accessibility folks to make sure default color contrast is sufficient, color choices aren't a problem for people with color vision deficiency, etc. They wouldn't even need to necessarily review the code to add value--they could evaluate the screenshots above. Evaluating the range of possibilities by reviewing the  code or experimenting with the result of this PR would be even better, of course.

We don't have a specific accessibility group as far as I know, but I'm going to guess that there might be people with that kind of knowledge on the website teams, so... @nodejs/website @nodejs/website-redesign

Trott commented:
Whoops, never mind if the colors have been removed? Sorry, everyone.

antsmartian commented:
@BridgeAR: I don't have any strong suggestion on the colors. But, can't we make it configurable with some config, so that the user can decide what color he/she wants to keep for the errors/stacktrace etc?

BridgeAR commented:
Landed in 6c40cb2aca89df4c7c0e3923d93024734dd49f2d :tada:
